test/TestCase.cpp

```diff
@@ -149,17 +149,27 @@ void EVMVersionRestrictedTestCase::processBytecodeFormatSetting()
 	// EOF only available since Osaka
 	solAssert(!eofVersion.has_value() || solidity::test::CommonOptions::get().evmVersion().supportsEOF());
 
-	std::string bytecodeFormatString = m_reader.stringSetting("bytecodeFormat", "legacy,>=EOFv1");
-	if (bytecodeFormatString == "legacy,>=EOFv1" || bytecodeFormatString == ">=EOFv1,legacy")
+	auto const bytecodeFormatStr = m_reader.stringSetting("bytecodeFormat", "");
+	if (bytecodeFormatStr == "legacy,>=EOFv1" || bytecodeFormatStr == ">=EOFv1,legacy")
+	{
+		m_bytecodeFormat.insert(BytecodeFormat::Legacy);
+		m_bytecodeFormat.insert(BytecodeFormat::EOFv1);
+	}
+	else if (bytecodeFormatStr == "legacy")
+		m_bytecodeFormat.insert(BytecodeFormat::Legacy);
+	else if (bytecodeFormatStr == ">=EOFv1")
+		m_bytecodeFormat.insert(BytecodeFormat::EOFv1);
+	else if (!bytecodeFormatStr.empty())
```

Comment:
Once we change the default back to `">=EOFv1,legacy"` as I suggested above, the empty value becomes useless because it will always just skip the test. It was not valid when the setting was a string, so I think we can keep it at that. It's not very useful.

Also, I'd just overwrite `m_bytecodeFormat` instead of adding to it in case this function is ever used on an already initialized object.

```suggestion
		m_bytecodeFormat = {BytecodeFormat::Legacy, BytecodeFormat::EOFv1};
	else if (bytecodeFormatStr == "legacy")
		m_bytecodeFormat = {BytecodeFormat::Legacy};
	else if (bytecodeFormatStr == ">=EOFv1")
		m_bytecodeFormat = {BytecodeFormat::EOFv1};
	else if (bytecodeFormatStr.empty())
		BOOST_THROW_EXCEPTION(std::runtime_error{"The 'bytecodeFormat' setting requires at least one value."});
	else
```